Core/Scripts Corrections on Halion's Encounter

sirikfoll opened the pull request:
**Changes proposed**:
(Everything that i did here was confirmed in retail 6.2.3.20886)
- Fixed the proportion of area scaling from Combustion and Consumption spells by the amount of stacks earned by the Marks of Combustion/Consumption(this was achieved adjusting the spell in spell_dbc)
- Normal Halion needs to be agressive and Twilight Halion needs to be Defensive
- Twilight Halion needs UNIT_FLAG_COMBAT just after summon, to avoid health regeneration(checked in sniff)
- Fixes summon of Living Embers(right timing, 3 seconds after Living Inferno spawn)
- Implement proper evade, by despawning both Halions and the controller.
- Fixed Blazing Aura issues
- Fixed radius of the damage from combustion and consumption
- Fixed all timers

**Target branch(es)**: 335/6x

**Issues addressed**: Fixes #16599 #9231 #16700 #16626 #15154

**Tests performed**: (Does it build, tested in-game, etc) Yes.

**Known issues and TODO list**: 
- [x] wanted to fix #15154 too, but dunno how, if anyone has an idea, please, share
- [x] check phase two-three transition(bug caused by me?)
- [x] sniff halion's Orientation after wipe
- [x] Blazing Aura issues #16626
- [x] timers
- [x] Twilight halion needs to run some events before aggro, check how to achiev that

Keader commented:
:+1:

Treeston commented:
Well, you know how this works.

While touching the boss script, you also get to sniff verify timers, spells et cetera and make sure it's hack free.

Enjoy!

sirikfoll commented:
I won't be able to do much for this one, complexity way beyond my capabilities '-'

ccrs commented:
also look into evading while at phase 2,3

Nawuko commented:
halion normal spawn pos:

[0] Position: X: 3156.67 Y: 533.8108 Z: 72.98822
[0] Orientation: 3.159046

halion after wipe spawn pos:

[1] Position: X: 3156.625 Y: 533.2674 Z: 72.97205
[1] Orientation: 0

Hope it helps :+1:

sirikfoll commented:
Thanks @Nawuko ^^

Keader commented:
#16626 Related with Halion Engage too :)

sirikfoll commented:
Any suggestions on #15154  ?
```
Fiery Combustion (74562)
This area affected by this spell every second will get bigger and bigger, the visual part works
but the radius of damage part isn't updated.
http://imgur.com/NWhGegd

joschiwald commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -212,115 +212,28 @@ CorporealityEntry const _corporealityReference[MAX_CORPOREALITY_STATE] = {
     {74831, 74836}
 };
 
-struct generic_halionAI : public BossAI
```
why did you removed this base class?

sirikfoll replied:
didn't liked it, in the end there's even more duplicated code using this class

Nyeriah replied:
What duplicated code was supposed to be gone with this removal? It seems to be adding duplicates rather than removing them

sirikfoll replied:
both bosses do not use the same timers for spells, that makes the genericAi useless

Warpten replied:
i will not merge this

sirikfoll replied:
well, feel free to close the PR or to commit some changes that looks right, I will not be changing this

Keader replied:
What's logic to maintain a script that is not useful? No make any sense for me

joschiwald commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -1819,6 +1832,40 @@ class spell_halion_spawn_living_embers : public SpellScriptLoader
+
+            void FilterTargets(std::list<WorldObject*>& targets)
+            {
+                targets.remove_if([this](WorldObject* target) { return target->GetEntry() != NPC_LIVING_EMBER; });
```
what about doing this with conditions?

sirikfoll replied:
doesn't work, there's already entrys in tdb

joschiwald replied:
it works if you do it right

Am 05.03.2016 um 21:02 schrieb Gustavo:
>
> In 
> src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp 
> <https://github.com/TrinityCore/TrinityCore/pull/16725#discussion_r55127036>:
>
> > @@ -1819,6 +1832,40 @@ class spell_halion_spawn_living_embers : public SpellScriptLoader
> >          }
> >  };
> >
> > +class spell_halion_blazing_aura : public SpellScriptLoader
> > +{
> > +    public:
> > +        spell_halion_blazing_aura() : SpellScriptLoader("spell_halion_blazing_aura") { }
> > +
> > +        class spell_halion_blazing_aura_SpellScript : public SpellScript
> > +        {
> > +            PrepareSpellScript(spell_halion_blazing_aura_SpellScript);
> > +
> > +            void FilterTargets(std::list<WorldObject*>& targets)
> > +            {
> > +                targets.remove_if([this](WorldObject* target) { return target->GetEntry() != NPC_LIVING_EMBER; });
>
> doesn't work, there's already entrys in tdb
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/TrinityCore/TrinityCore/pull/16725/files#r55127036>.
>

sirikfoll replied:
alright, then i dunno how, open to suggestions :)

joschiwald replied:
paste your conditions here to see whats wrong

sirikfoll replied:
issue adressed, it needs conditions for both difficulties of the spell, i was using only for the first ^^

Runico commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -388,18 +303,44 @@ class boss_halion : public CreatureScript
+                DoMeleeAttackIfReady();
             }
 
             void ExecuteEvent(uint32 eventId) override
```
is this function requires override ?

sirikfoll replied:
BossAI::UpdateAI  update events even if it has UNIT_STATE_CASTING, i'm having a hard time trying to understand how the cooldowns need to be handled, but that still may be needed

sirikfoll replied:
now it can be moved to inside UpdateAi, I will do that later ^^

tkrokli commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -546,27 +494,53 @@ class boss_twilight_halion : public CreatureScript
-                            controller->AI()->DoAction(ACTION_MONITOR_CORPOREALITY);
-                        break;
-                    default:
-                        generic_halionAI::SpellHit(who, spell);
```
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:493:33: fatal error: 
```cpp
      unused parameter 'who' [-Wunused-parameter]

            void SpellHit(Unit* who, SpellInfo const* spell) override
                                ^
```
You need to look at line 493, which is 9 lines above this one, `generic_halionAI::SpellHit(who, spell);`

Treeston commented:
@sirikfoll look at how [Grobbulus does it](https://github.com/Treeston/TrinityCore/blob/3.3.5-naxxcleanup/src/server/scripts/Northrend/Naxxramas/boss_grobbulus.cpp#L185-L221), it's the same mechanic.

sirikfoll commented:
interesting, I was trying with https://gist.github.com/sirikfoll/b73c464392746dbe4011, probally just needs to create a formula then, thanks ^^

Treeston commented:
If it actually has a stacking aura that tracks the progress, then that's the proper way to do it (and grobb is a hack). I haven't looked into it, admittedly.

sirikfoll commented:
Alright, this is it, up to reviewing and testing.

ps: I will not be trying to fix corporeality, leave that for another brave soul

Shauren commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -626,21 +588,36 @@ class npc_halion_controller : public CreatureScript
                 _instance->SetBossState(DATA_HALION, FAIL);
+                _summons.DespawnAll();
+
+                uint32 corpseDelay = me->GetCorpseDelay();
```
use _DespawnAtEvade, now available in 3.3.5 branch

sirikfoll replied:
this npc doesn't use BossAI, dunno how to call it from here

Treeston commented:
While you're touching the boss already, mind switching all timers to use `std::chrono` types? See #16524 for plenty of examples, the code you get is way cleaner.

Keader commented:
Halion casts (Flame / Dark Breath) can be interrupted. #12020
Even though not related directly with this PR, it is part of the fight.
In reality, all Halion spells can't be interrupted...
I believe that the simplest solution (at least temporarily) is:

```sql
UPDATE `creature_template` SET `mechanic_immune_mask`=`mechanic_immune_mask` | 33554432 WHERE `entry` IN (39863,39864,39944,39945);
UPDATE `creature_template` SET `mechanic_immune_mask`=`mechanic_immune_mask` | 33554432 WHERE `entry` IN (40142,40143,40144,40145);
```
Add  interrupt immunity on Halion and Twlight Halion.

Warpten commented:
I'll check this out after work.

Keader commented:
@Warpten if you have time after this, can you revise damage mods on Ruby Sanctum? Trash mobs has no "white damage" and Halion and Mini-bosses seems have wrong damage (low damages).

sirikfoll committed:
Core/Scripts Corrections on Halion's Encounter

Fixed the proportion of area scaling from Combustion and Consumption spells by the amount of stacks earned by the Marks of Combustion/Consumption(this was achieved adjusting the spell in spell_dbc)

Normal Halion needs to be agressive and Twilight Halion needs to be Defensive

Twilight Halion needs UNIT_FLAG_COMBAT just after summon, to avoid health regeneration(checked in sniff)

Fixes summon of Living Embers(right timing, 3 seconds after Living Inferno spawn)

Implement proper evade, by despawning both Halions and the controller.

Fixed Blazing Aura issues

Fixed radius of the damage from combustion and consumption

Fixed all timers

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -470,26 +396,29 @@ class boss_twilight_halion : public CreatureScript
                 halion->AddAura(SPELL_COPY_DAMAGE, me); // We use explicit targeting here to avoid conditions + SPELL_ATTR6_CANT_TARGET_SELF.
                 me->AddAura(SPELL_COPY_DAMAGE, halion);
-                me->AddAura(SPELL_DUSK_SHROUD, me);
+                DoCast(me, SPELL_DUSK_SHROUD, true);
```
no spell cast was seen in sniffs

sirikfoll replied:
https://gist.github.com/sirikfoll/84243ce63b596240c133

Warpten replied:
these two are hidden, you see the copy damage one in spell_non_melee_damage_log and tw_precision in aura updates (at least in 335)

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -212,132 +213,39 @@ CorporealityEntry const _corporealityReference[MAX_CORPOREALITY_STATE] = {
     {74831, 74836}
 };
 
-struct generic_halionAI : public BossAI
```
then you just override EnterCombat in children AIs and set the timers there (after calling parent's)

sirikfoll replied:
I will not restore this

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -212,132 +213,39 @@ CorporealityEntry const _corporealityReference[MAX_CORPOREALITY_STATE] = {
 
             void EnterEvadeMode(EvadeReason why) override
             {
-                if (why == EVADE_REASON_BOUNDARY)
```
this is new. does it even work? the way i remember it door boundary won't work if door is spawned where the boss is - which is the case herz

sirikfoll replied:
so far I wasn't able to successfully remove him from the circle, the boundary seems  very sensitive, he always despawn before leaving the circle ^^

Nawuko replied:
Boundary is checked on the player position. Not on the position of halion. Thats why it feels very sensitive

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -470,26 +396,29 @@ class boss_twilight_halion : public CreatureScript
                 me->SetHealth(halion->GetHealth());
                 me->SetPhaseMask(0x20, true);
-                me->SetReactState(REACT_AGGRESSIVE);
+                me->SetReactState(REACT_DEFENSIVE);
```
Would need to check again, but there's a reason that healers should not enter the twilight realm first: the boss would 1shot them. This change cancels that, the only threat *cough* being Twilight Cutter

sirikfoll replied:
Everything here was sniffed and tested by me in retail, he is defensive. I have it in video too(that I rather not be need to upload '-')

sirikfoll replied:
Some proofs:
https://youtu.be/MTPb4C2gLZk?t=99
https://youtu.be/VAR1olGzApI?t=1m58s

Warpten replied:
from second video looks like hes more or less pulsing aggro in some range

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -470,26 +396,29 @@ class boss_twilight_halion : public CreatureScript
                 me->SetPhaseMask(0x20, true);
-                me->SetReactState(REACT_AGGRESSIVE);
+                me->SetReactState(REACT_DEFENSIVE);
+                me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IN_COMBAT);
```
this is a hack, that flag is set in db but the core removes it since threat list is empty on spawn.

sirikfoll replied:
Then the DB flag is a hack.
He is spawned with [14] UNIT_FIELD_FLAGS: 32832/4.600743E-41
right next comes a SMSG_UPDATE_OBJECT packet with [0] UNIT_FIELD_FLAGS: 557120/7.806914E-40
is exactaly as it should be

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -470,26 +396,29 @@ class boss_twilight_halion : public CreatureScript
+                _EnterCombat();
+                me->AddAura(SPELL_TWILIGHT_PRECISION, me);
+                events.ScheduleEvent(EVENT_CLEAVE, Seconds(3));
+                events.ScheduleEvent(EVENT_BREATH, Seconds(0));
```
Breath on engage?

sirikfoll replied:
Everything here was sniffed and tested by me in retail, while "out of combat" he already uses Tail Slash and Consumption, and yes, breath on engage.

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -522,6 +451,9 @@ class boss_twilight_halion : public CreatureScript
             void DamageTaken(Unit* attacker, uint32& damage) override
             {
+                if (!events.IsInPhase(PHASE_TWO) && !events.IsInPhase(PHASE_THREE))
+                    EnterCombat(attacker);
```
this shouldn't be needed

sirikfoll replied:
this became needed because of the UNIT_FLAG_IN_COMBAT, the EnterCombat hook will never be called unless I explicitly do it.

Warpten replied:
document it in source then

sirikfoll replied:
done

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -542,33 +474,56 @@ class boss_twilight_halion : public CreatureScript
-                    default:
-                        generic_halionAI::SpellHit(who, spell);
-                        break;
+                case SPELL_TWILIGHT_DIVISION:
```
did the tc codestyle guidelines change?

sirikfoll replied:
indeed my bad, VS always format it this way and i keep forgetting, sorry

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -1473,6 +1491,47 @@ class spell_halion_combustion_consumption : public SpellScriptLoader
+                    return;
+
+                uint32 triggerSpell = GetSpellInfo()->Effects[aurEff->GetEffIndex()].TriggerSpell;
+                int32 radius = caster->GetObjectScale() * M_PI * 10000 / 3;
```
where does this come from?

sirikfoll replied:
nothing, trial and error until you find something that fits the visual range of the spell

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -1819,6 +1878,35 @@ class spell_halion_spawn_living_embers : public SpellScriptLoader
+            void HandleScript(SpellEffIndex effIndex)
+            {
+                PreventHitDefaultEffect(effIndex);
+                GetHitUnit()->CastSpell(GetHitUnit(), GetSpellInfo()->Effects[EFFECT_1].TriggerSpell);
```
why ?

sirikfoll replied:
Well, this is the only way I could find to make ir work and there's similar things over the core, open to suggestions tho

Warpten replied:
My point is that its already a force cast on all targets hit, this shouldnt be needed, sounds like a spellsystem bug?

sirikfoll replied:
this I dunno

sirikfoll commented:
My main concern here is that somehow Halion could be summoned twice, It has happened once(while in the middle of work), but in the final version, didn't saw it anymore. Anyone see a possible flaw that could cause such a thing?

sirikfoll commented:
Hey, should Living Embers really Enrage, I didn't found a source for that. Anyone remembers?

Keader commented:
I never see Living Ember cast Berserk...
In my sniff, i cant find any reference about it....
@Nawuko if you have sniff, can you check?

Nawuko commented:
I dont have sniffs with berserk on Embers. Only http://wotlk.openwow.com/spell=75887 But no enrage

Warpten commented:
can you get halion to cast consumption/combustion on the tank if theres only 1 player in instance? i clearly remember this not being the case but i went to check on retail and they changed that.

Keader commented:
Last few times that i did Halion on blizzard and he never cast combustion / consuption in tanker... (On MOP and WoD patchs)

Warpten commented:
i just went there and he cast it on me.

Keader commented:
rly? i will login to check it now

Keader commented:
@Warpten With Mage Class without pets, he dont give me combustion/consuption on retail.
If your pet is tanker... he give you combustion/consuption normally.

Demonid commented:
@sirikfoll You mean like this: http://puu.sh/osSuK/5f638a9e7a.jpg

I added the PR on a live server to test it, everything seems fine but this happened, then again the Respawn of the boss after a wipe takes some time.

ccrs commented:
add a check to instanceScript on OnCreatureCreate

Keader commented:
@Demonid both Halion has script?

Demonid commented:
@Keader I dont actually know, a player reported that, i believe it happens after a wipe sometimes.

Btw, this pr is no longer being worked right?

sirikfoll commented:
I dunno how to reproduce the double spawn, neither what is the cause, I can add few more checks to try preventing it, but ideally we should discover the root of the problem

Nawuko commented:
Hey. I just want to throw that in here. https://gist.github.com/Nawuko/5447cd31c08054ddb0009b56e7087d61 Our updated version with your spells/timings :+1: 

Not perfect but hey, maybe you can use 1-2 things from there.

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -889,52 +873,74 @@ class npc_orb_carrier : public CreatureScript
+                _instance(creature->GetInstanceScript())
             {
                 ASSERT(creature->GetVehicleKit());
+                me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
```
? It should have the invisible trigger extra flag, so that shouldnt be needed ? You do see it in .gm on but that's intended

sirikfoll committed:
Fix Halion double summon

Re-Adjust Twilight Halion breath timer, to match 3.3.5
Add trigger flag to Npc orb Carrier

sirikfoll commented:
Well, I think that the double summon problem is solved now

Demonid commented:
@sirikfoll What about this bug:

1- Go into Phase 3
2- Use Haunt (Afliction Warlock) on the Crepuscular Halion
3- Wipe (Kill yourself by any means)

The encounter will stuck in combat for ever.

sirikfoll commented:
Haven't been able to reproduce yet, is your core updated? I only had evade issues on older versions.
Also, is Combustion periodic damage with correct range in 25 heroic difficulty? Everything is the same, but 25h seems not to be working.

Demonid commented:
Weird, im using this rev: 204acbb5b72d9706cdcf26aede62c2eeec11c6a2 with some custom code on ICC, going to test again with a clean source + this Pr.

sirikfoll commented:
I will investigate more, thanks a lot for the feedback :D

sirikfoll commented:
@Demonid  I saw this combat bug and the cause, at least for me, is that Halion Controller(the trigger) is getting threat with both Halions. Therefore, they keep each other from evading.
I am unable to reproduce it by myself, only happens on the live server, which I can't use to debug. 
So I have no idea of why this is happening and how to prevent it. 
Might as well be a generic bug.

sirikfoll committed:
Halion - A few improvements

Add missing UpdateVictim() in Twilight Halion
Living Embers now will properly enter in combat
Remove enrage from Living Embers(no one could confirm that this enrage is blizzlike)

sirikfoll committed:
Halion - dbc data fixes

Combustion and Consumption Heroic versions lacks radius data
Consumption 25 heroic version has Mechanic data swapped, the effect needs to be dispellable, like the 10 heroic version.

sirikfoll committed:
Halion - Meteor Strike flames now will be shaped correctly

Shape between and X and a cross

sirikfoll commented:
The evade issues is not this PR's fault, is a generic problem #17180.

@Demonid here is a for the evade problem, won't be added to the PR because is considered a hack, but it works perfectly: https://gist.github.com/sirikfoll/0b28b8cd06ac90eb8876fb4bade4f8db

Aokromes commented:
Any reason to don't merge this?

sirikfoll committed:
NULL -> nullptr changes

Also correct timer for Twilight cutter

sirikfoll commented:
at least at my look, it's finished.

ccrs commented:
sometimes heroic spawns from the cutter orbs remain stationary instead of moving in circles after a wipe, have u checked that?

sirikfoll commented:
Its rare, setActive(true), on the orb carrier solve that, but i'm pretty sure devs won't accept that

Treeston commented:
`SetActive(true)` on engage and `SetActive(false)`reset is fine by me. BossAI does it too.

sirikfoll commented:
@Treeston Since I don't know the cause to this problem, and the orbs are pathing all the time, even when there's no boss in there, i can't confirm if that will solve the problem.
I just tested with a setActive in the constructor;

ccrs commented:
its usually related to creatures entering combat, check that.

sirikfoll commented:
@ccrs  If you know how to reproduce the Shadow orbs problem, can you, please, test if this solve the issue? https://gist.github.com/sirikfoll/c3675e6d82a20edaef1f62bdce17b6a3

ccrs commented:
Its quite random.. try to engage and reset till it happens.

Aokromes commented:
Maybe this can be merged and the remain issues fixed on the future?

Treeston commented:
We don't merge PRs that introduce new known issues.

sirikfoll commented:
@Treeston What "new known issues" this pr introduces?
The problem with evade and with the Shadow Orbs are old things, and I presented solutions for both, only waiting for you guys decide if it can be used or not.

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -1006,10 +1024,7 @@ class npc_meteor_strike_initial : public CreatureScript
                         Position newPos = me->GetNearPosition(10.0f, 0.0f); // Exact distance
                         if (Creature* meteor = me->SummonCreature(NPC_METEOR_STRIKE_NORTH + i, newPos, TEMPSUMMON_TIMED_DESPAWN, 30000))
-                        {
-                            meteor->SetOrientation(angle[i]);
```
I never tested those changes i did with the random adjusting because my pc is way too shitty but I clearly remember that this line was needed for the fire spread to go in the correct direction and not randomly fuck over the terrain. Even though that's what it should be, orientation is used to determine the randomness in the position of the next spawn, so i'm not sure why this is gone?

sirikfoll replied:
Don't ask me why, i don't understood the code, it was basicaly trial and error, but look at the results, shouldn't be like this?
http://prntscr.com/bkkl0s
http://prntscr.com/bkkkl7
http://prntscr.com/bkkkda
http://prntscr.com/bkkkl7
That `static_cast<float>(M_PI / 5.0f), static_cast<float>(M_PI / 2.0f)` makes this happen, the fire shapes between a cross and a X shape

Warpten commented on src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:
```diff
@@ -987,12 +1005,12 @@ class npc_meteor_strike_initial : public CreatureScript
                 {
                     Position const* ownerPos = halionAI->GetMeteorStrikePosition();
-                    float randomAdjustment = frand(0.0f, static_cast<float>(M_PI / 5.0f));
+                    float randomAdjustment = frand(static_cast<float>(M_PI / 5.0f), static_cast<float>(M_PI / 2.0f));
```
This is forcing the fire to always spread left, even though it can go both directions

Runico commented:
@sirikfoll Could you confirm, Halion ( 25Man Mode ) should always drop a trinket,

http://www.truewow.org/forum/viewtopic.php?t=13613

sirikfoll commented:
@Runico  could be true, but I don't even know how to change the loot

Runico commented:
@sirikfoll these sql works, but we just need confirm this is blizlike or not


```SQL

DELETE FROM creature_loot_template WHERE entry IN(39864, 39945);
INSERT INTO creature_loot_template VALUES
(39864, 1, 100, 1, 0, -34282, 2),
(39864, 2, 100, 1, 0, -45100, 1),
(39864, 49426, 100, 1, 0, 3, 3),
(39945, 1, 100, 1, 0, -34283, 2),
(39945, 2, 100, 1, 0, -45101, 1),
(39945, 49426, 100, 1, 0, 3, 3);

DELETE FROM reference_loot_template WHERE entry IN(34282, 34283, 45100, 45101);
INSERT INTO reference_loot_template VALUES
(34282, 53125, 0, 1, 1, 1, 1),
(34282, 53126, 0, 1, 1, 1, 1),
(34282, 53127, 0, 1, 1, 1, 1),
(34282, 53129, 0, 1, 1, 1, 1),
(34282, 53132, 0, 1, 1, 1, 1),
(34282, 53133, 0, 1, 1, 1, 1),
(34282, 53134, 0, 1, 1, 1, 1),
(34282, 53486, 0, 1, 1, 1, 1),
(34282, 53487, 0, 1, 1, 1, 1),
(34282, 53488, 0, 1, 1, 1, 1),
(34282, 53489, 0, 1, 1, 1, 1),
(34282, 53490, 0, 1, 1, 1, 1),
(45100, 54569, 0, 1, 1, 1, 1),
(45100, 54571, 0, 1, 1, 1, 1),
(45100, 54572, 0, 1, 1, 1, 1),
(45100, 54573, 0, 1, 1, 1, 1),
(34283, 54576, 0, 1, 1, 1, 1),
(34283, 54587, 0, 1, 1, 1, 1),
(34283, 54586, 0, 1, 1, 1, 1),
(34283, 54585, 0, 1, 1, 1, 1),
(34283, 54584, 0, 1, 1, 1, 1),
(34283, 54583, 0, 1, 1, 1, 1),
(34283, 54582, 0, 1, 1, 1, 1),
(34283, 54581, 0, 1, 1, 1, 1),
(34283, 54580, 0, 1, 1, 1, 1),
(34283, 54579, 0, 1, 1, 1, 1),
(34283, 54578, 0, 1, 1, 1, 1),
(34283, 54577, 0, 1, 1, 1, 1),
(45101, 54591, 0, 1, 1, 1, 1),
(45101, 54590, 0, 1, 1, 1, 1),
(45101, 54589, 0, 1, 1, 1, 1),
(45101, 54588, 0, 1, 1, 1, 1);

```
